core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -405,29 +454,62 @@ export class OptimizedSsrEngine {
         }
         this.log(
           `Rendering of ${request?.originalUrl} was not able to complete. This might cause memory leaks!`,
-          false
+          false,
+          { request }
         );
       }, this.ssrOptions?.maxRenderTime ?? 300000); // 300000ms == 5 minutes
 
-    this.log(`Rendering started (${request?.originalUrl})`);
+    this.log(`Rendering started (${request?.originalUrl})`, true, { request });
     this.renderingCache.setAsRendering(renderingKey);
     this.currentConcurrency++;
 
+    options = {
+      ...options,
+      providers: [
+        {
+          provide: EXPRESS_SERVER_LOGGER,
+          useValue: this.logger,
+        },
+      ],
+    };
+
     this.expressEngine(filePath, options, (err, html) => {
       if (!maxRenderTimeout) {
         // ignore this render's result because it exceeded maxRenderTimeout
         this.log(
           `Rendering of ${request.originalUrl} completed after the specified maxRenderTime, therefore it was ignored.`,
-          false
+          false,
+          { request }
         );
         return;
       }
       clearTimeout(maxRenderTimeout);
 
-      this.log(`Rendering completed (${request?.originalUrl})`);
+      this.log(`Rendering completed (${request?.originalUrl})`, true, {
+        request,
+      });
       this.currentConcurrency--;
 
       renderCallback(err, html);
     });
   }
+
+  //CXSPA-3680 - remove this method in 7.0
```

Comment:
[MINOR]

Should we drop this whole method or change it in 7.0?

```suggestion
  //CXSPA-3680 - in a new major version let's use `DefaultExpressServerLogger` by default. And drop `LegacyExpressServerLogger`
```

Reply:
My plan was to set `DefaultExpressServerLogger` as default in `defaultSsrOptimizationOptions`. Thanks to that it will be used unless we provide custom longer as part of options. In that case, we don't need a method to initialize the logger.